src/PointwiseFunctions/Hydro/EquationsOfState/Tabulated3d.hpp

```diff
@@ -0,0 +1,207 @@
+                  double enthalpy_minimum);
+
+  bool is_equal(const EquationOfState<IsRelativistic, 3>& rhs) const override;
+
+  bool operator==(const Tabulated3D<IsRelativistic>& rhs) const;
+
+  bool operator!=(const Tabulated3D<IsRelativistic>& rhs) const;
+
+  template <typename DataType>
+  void enforce_physicality(Scalar<DataType>& electron_fraction,
+                           Scalar<DataType>& density,
+                           Scalar<DataType>& temperature) const;
+
+  WRAPPED_PUPable_decl_base_template(  // NOLINT
+      SINGLE_ARG(EquationOfState<IsRelativistic, 3>), Tabulated3D);
+
+
+  /// The lower bound of the electron fraction that is valid for this EOS
+  double electron_fraction_lower_bound() const override {
+    return table_electron_fraction_.front();
+  }
+
+  /// The upper bound of the electron fraction that is valid for this EOS
+  double electron_fraction_upper_bound() const override {
+    return table_electron_fraction_.back();
+  }
+
+  /// The lower bound of the rest mass density that is valid for this EOS
+  double rest_mass_density_lower_bound() const override {
+    return std::exp((table_log_density_.front()));
+  }
+
+  /// The upper bound of the rest mass density that is valid for this EOS
+  double rest_mass_density_upper_bound() const override {
+    return std::exp((table_log_density_.back()));
+  }
+
+  /// The lower bound of the temperature that is valid for this EOS
+  double temperature_lower_bound() const {
+    return std::exp((table_log_temperature_.front()));
+  }
+
+  /// The upper bound of the temperature that is valid for this EOS
+  double temperature_upper_bound() const {
+    return std::exp((table_log_temperature_.back()));
+  }
+
+  /// The lower bound of the specific internal energy that is valid for this EOS
+  /// at the given rest mass density \f$\rho\f$ and electron fraction \f$Y_e\f$
+  double specific_internal_energy_lower_bound(
+      const double rest_mass_density,
+      const double electron_fraction) const override;
+
+  /// The upper bound of the specific internal energy that is valid for this EOS
+  /// at the given rest mass density \f$\rho\f$
+  double specific_internal_energy_upper_bound(
+      const double rest_mass_density,
+      const double electron_fraction) const override;
+
+  /// The lower bound of the specific enthalpy that is valid for this EOS
+  double specific_enthalpy_lower_bound() const override {
+    return enthalpy_minimum_;
+  }
+
+ private:
+  EQUATION_OF_STATE_FORWARD_DECLARE_MEMBER_IMPLS(3)
+
+  /// Energy shift used to account for negative specific internal energies,
+  /// which are only stored logarithmically
+  double energy_shift_ = 0.;
+
+  /// Enthalpy minium  across the table
+  double enthalpy_minimum_ = 1.;
+
+  /// Main interpolator for the EoS.
+  /// The ordering is  \f$(\log T. \log \rho, Y_e)\f$.
+  intrp::UniformMultiLinearSpanInterpolation<3, NUM_TAB3D_VARS> interpolator_{};
+  /// Electron fraction
```

Comment:
Can you add `. Assumed to be sorted in ascending order.` to these? Since `specific_internal_energy_upper_bound` etc. assume that